src/foam/u2/view/agreementView.js

```diff
@@ -0,0 +1,56 @@
+foam.CLASS({
+  package: 'foam.u2.view',
+  name: 'agreementView',
+  extends: 'foam.u2.Controller',
+
+  imports: [
+      'fileDAO'
+    ],
+
+  css: `
+    ^text {
+        overflow-y: scroll;
+        height: 500px;
+        width: 100%;
+        border: 1px solid #DDD;
+        padding: 10px;
+    }
+
+    ^pdf embed {
+      width: 100%;
+      height: 500px;
+    }
+  `,
+
+  properties: [
+    {
+      name: 'fileId'
+    }
+  ],
+
+  methods: [
+    async function initE() {
+      this.SUPER();
+      let file = await ctrl.__subContext__.fileDAO.find(this.fileId);
```

Comment:
Why accessing subContext instead of just importing and accessing as this.fileDAO?

Reply:
I'am getting an error, that fileDAO is undefined if just import it and use this.fileDAO

Reply:
That means that you created the view without providing the context. Where is it used?

Reply:
BaseAcceptanceDocumentCapability.js
Now I see where the error. Thank you.